### PR TITLE
Serialization test

### PR DESCRIPTION
**Pull request by fluidvanadium**
this test shows a production is unexpected. i.e. it exists under unsure circumstances

### CODE REVIEW

**Comment by fluidvanadium** [comment]
builds atop https://github.com/zingolabs/zingolib/pull/804

File: zingoconfig/src/lib.rs [hunk 1]
```diff
@@ -116,12 +116,38 @@ pub struct ZingoConfig {
...
+        self.lightwalletd_uri = Some(lightwalletd_uri);
         self
     }
-    pub fn set_lightwalletd(mut self, lightwalletd_uri: http::Uri) -> Self {
-        self.lightwalletd_uri = Some(lightwalletd_uri);
+    /// # Examples
+    /// ```
+    /// use zingoconfig::ZingoConfigBuilder;
+    /// use zingoconfig::ChainType::Testnet;
+    /// assert_eq!(ZingoConfigBuilder::default().set_chain(Testnet).create().chain, Testnet);
```
Comment (zancas) [inline zingoconfig/src/lib.rs:138]: Brilliant!

File: zingoconfig/src/lib.rs [hunk 2]
```diff
@@ -116,12 +116,38 @@ pub struct ZingoConfig {
...
 }
 impl ZingoConfigBuilder {
-    pub fn set_wallet_dir(mut self, dir: PathBuf) -> Self {
-        self.wallet_dir = Some(dir);
+    /// the proxy server we download blockchain information from
+    /// # Examples
+    /// ```
+    /// use zingoconfig::ZingoConfigBuilder;
+    /// use http::Uri;
+    /// assert_eq!(ZingoConfigBuilder::default().set_lightwalletd_uri(("https://zcash.mysideoftheweb.com:19067").parse::<Uri>().unwrap()).lightwalletd_uri.clone().unwrap(), "https://zcash.mysideoftheweb.com:19067");
```
Comment (zancas) [inline zingoconfig/src/lib.rs:124]: UtAck.  Lovely!

File: integration-tests/tests/integrations.rs [hunk 3]
```diff
@@ -3585,6 +3435,145 @@ mod slow {
...
         assert_eq!(bala_sim, bala_syn);
     }
+    async fn load_wallet_from_data_and_assert(
+        data: &[u8],
+        expected_balance: u64,
+        num_addresses: usize,
+    ) {
+        let config = zingoconfig::ZingoConfig::build(ChainType::Testnet)
+            .set_lightwalletd_uri(
+                ("https://zcash.mysideoftheweb.com:19067")
```
Comment (AloeareV) [inline integration-tests/tests/integrations.rs:3445]: Ooh, working testnet server?

Comment (AloeareV) [reply integration-tests/tests/integrations.rs:3445]: Why do we need a lwd uri at all? Testing wallet reading and writing shouldn't need a network connection at all

File: zingoconfig/src/lib.rs [hunk 4]
```diff
@@ -116,12 +116,38 @@ pub struct ZingoConfig {
...
+    /// ```
+    /// use zingoconfig::ZingoConfigBuilder;
+    /// use zingoconfig::ChainType::Testnet;
+    /// assert_eq!(ZingoConfigBuilder::default().set_chain(Testnet).create().chain, Testnet);
+    /// ```
+    pub fn set_chain(&mut self, chain: ChainType) -> &Self {
+        self.chain = chain;
+        self
+    }
+    /// the proxy server we download blockchain information from
```
Comment (zancas) [inline zingoconfig/src/lib.rs]: This is just copied from elsewhere and is incorrect.

**Review by zancas — CHANGES_REQUESTED** [review]
@fluidvanadium please merge this code into your branch to get my fixes:

https://github.com/fluidvanadium/zingolib/pull/9

**Comment by zancas** [comment]
@AloeareV when you reviewed this code, I assume you ran the tests locally?

**Comment by AloeareV** [comment]
> @AloeareV when you reviewed this code, I assume you ran the tests locally?

I didn't get that far...I had changes to request before I got to it!

File: zingoconfig/src/lib.rs (hunk 4 above)
Comment (fluidvanadium) [reply zingoconfig/src/lib.rs]: fixed

File: integration-tests/tests/integrations.rs (hunk 3 above)
Comment (fluidvanadium) [reply integration-tests/tests/integrations.rs:3445]: only for testing that it sends

**Pull request merged**